src/main/java/hudson/plugins/promoted_builds/conditions/DownstreamPassCondition.java

```diff
@@ -101,47 +90,89 @@ public PromotionBadge isMet(PromotionProcess promotionProcess, AbstractBuild<?,?
                         }
                     }
                 }
+
+                if (pdb!=null) {// if fingerprint doesn't have any, try the pseudo-downstream
+                    for (AbstractBuild<?,?> b : pdb.listBuilds(j)) {
+                        if (!b.isBuilding()) {
+                            Result r = b.getResult();
+                            if ((r == Result.SUCCESS) || (evenIfUnstable && r == Result.UNSTABLE)) {
+                                badge.add(b);
+                                continue OUTER;
+                            }
+                        }
+                    }
+                }
+
+                // none of the builds of this job passed.
+                return null;
             }
 
-            // none of the builds of this job passed.
-            return null;
-        }
-        
         return badge;
     }
 
     /**
+     *  @deprecated use {@link #getJobList(hudson.model.ItemGroup, hudson.model.AbstractProject, hudson.EnvVars)} 
      * List of downstream jobs that we need to monitor.
      *
      * @return never null.
      */
-    public List<AbstractProject<?,?>> getJobList(ItemGroup context) {
+    public List<AbstractProject<?,?>> getJobList(ItemGroup context){
+        return getJobList(context, null);
+    }
+    /**
+     * 
+     * List of downstream jobs that we need to monitor resolving variables.
+     * @since 2.33
+     * @return never null.
+     */
+    public List<AbstractProject<?,?>> getJobList(ItemGroup context, EnvVars buildEnvironment) {
         List<AbstractProject<?,?>> r = new ArrayList<AbstractProject<?,?>>();
-        for (String name : Util.tokenize(jobs,",")) {
+
+        String expandedJobs = getExpandedJobs(jobs, buildEnvironment);
```

Comment:
This change seems to be a security risk, because you allow injecting arbitrary job names.